site/docs/develop/java.md

```diff
@@ -207,21 +207,40 @@ api.transplantCommitsIntoBranch()
 Nessie has multiple `NessieAuthenticationProvider` implementations that allow different client authentication mechanisms as can be seen below.
 The documentation for how to configure Nessie server authentication can be found [here](../try/authentication.md).
 
-The `BasicAuthenticationProvider` allows connecting to a Nessie server that has `BASIC` authentication enabled.
-Note that `BASIC` is not supported in production and should only be used for development/testing.
-```java
-NessieApiV1 api =
-  HttpClientBuilder.builder()
-  .withUri(URI.create("http://localhost:19120/api/v1"))
-  .withAuthentication(BasicAuthenticationProvider.create("my_username", "very_secret"))
-  .build(NessieApiV1.class);
-```
-
-The `BearerAuthenticationProvider` allows connecting to a Nessie server that has `BEARER` authentication enabled.
-```java
-NessieApiV1 api =
-  HttpClientBuilder.builder()
-  .withUri(URI.create("http://localhost:19120/api/v1"))
-  .withAuthentication(BearerAuthenticationProvider.create("bearerToken"))
-  .build(NessieApiV1.class);
-```
+When configured with authentication enabled, a Nessie server expects every HTTP request to contain a 
+valid Bearer token in an `Authorization` header. Two authentication providers allow a Nessie client
+to automatically add the required token to the HTTP requests:
+
+1. The `BearerAuthenticationProvider` is the simplest one and directly takes the Bearer token as a 
+parameter; _the token must be valid for the entire duration of the client's lifetime_:
+
+    ```java
+    NessieApiV2 api =
+      NessieClientBuilder.createClientBuilder(null, null)
+      .withUri(URI.create("http://localhost:19120/api/v2"))
+      .withAuthentication(BearerAuthenticationProvider.create("bearerToken"))
+      .build(NessieApiV2.class);
+    ```
+
+2. The `Oauth2AuthenticationProvider` is more elaborate; at a minimum, it takes an OAuth2 token 
+endpoint URI, a Client ID and a Client Secret, and uses them to obtain an access token from the 
+token endpoint, which is then used as a Bearer token to authenticate against Nessie:
+
+    ```java
+    Map<String, String> authConfig =
+        Map.of(
+            CONF_NESSIE_AUTH_TYPE, "OAUTH2",
+            CONF_NESSIE_OAUTH2_TOKEN_ENDPOINT,
+                "https://<oidc-server>/realms/<realm-name>/protocol/openid-connect/token",
+            CONF_NESSIE_OAUTH2_CLIENT_ID, "my_client_id",
+            CONF_NESSIE_OAUTH2_CLIENT_SECRET, "very_secret");
+    NessieApiV2 api =
```

Comment:
I noticed incidentally when creating this example that the programmatic configuration of the OAuth2 provider is convoluted. There is no easy factory method available, and `OAuth2Client` and `OAuth2ClientParams` are package-private.

Should we improve that? (In a separate PR, of course.)